### ISSUE TITLE
bug: clang collision with home-manager (22.05)

### ISSUE THREAD

**Issue by notfed** [nix-community/home-manager#2997]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Issue description

It seems the package `clang` (and `gcc`; see next comment) collides with  `home-manager`. I added `clang` to the `home.packages` section of my `home.nix` file, and ran `home-manager switch`.

I got this collision error:

```
error: collision between `/nix/store/2b99rpx8zwdjjqkrvk7kqgn9mxhiidjy-binutils-2.38/bin/strip' and `/nix/store/r2h2gkjsr0xmxk1s0bx6p4v2czpc3km8-clang-wrapper-11.1.0/bin/strip'
error: builder for '/nix/store/gc69cdh6j70igfwln7qfrcy2rrmcr88j-home-manager-path.drv' failed with exit code 25;
       last 1 log lines:
       > error: collision between `/nix/store/2b99rpx8zwdjjqkrvk7kqgn9mxhiidjy-binutils-2.38/bin/strip' and `/nix/store/r2h2gkjsr0xmxk1s0bx6p4v2czpc3km8-clang-wrapper-11.1.0/bin/strip'
       For full logs, run 'nix log /nix/store/gc69cdh6j70igfwln7qfrcy2rrmcr88j-home-manager-path.drv'.
error: 1 dependencies of derivation '/nix/store/cfcx88rnx99r2j6nc8xcn88556lmybg4-home-manager-generation.drv' failed to build
```

I interpret this to mean that `clang` is trying to bring the executable `bin/strip` into scope, but the executable is already in my PATH from another package. And, interestingly, with the help of `nix-tree`, I realized the conflicting package is the `home-manager` package itself.

(EDIT: Re-titled the issue, and removed some red herring text: originally this was deadlocking me from re-installing `home-manager`, but I solved this with `nix-shell -p home-manager`, then declaratively uninstalling `clang`. I still can't install `clang`.)

### Maintainer CC

@dtzWill 
@Ericson2314 

### System information

```markdown
nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.17.7-zen1, NixOS, 22.05 (Quokka), 22.05.209.d1086907f56`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.8.1`
 - channels(jay): `"home-manager-22.05.tar.gz, nixos-22.05, nixos-21.05-21.05"`
 - channels(root): `"nixos-22.05.209.d1086907f56"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by notfed** [comment]
I also can't install `gcc` declaratively through `home.nix`. Same problem.

```
$ home-manager -f home.nix switch
these 3 derivations will be built:
  /nix/store/56al72sri1bv11rg858m2f9s17j9sb6x-home-manager-path.drv
  /nix/store/sfl7sh88w84hjra7pc3ahfcg4rcq8v9v-activation-script.drv
  /nix/store/4fwfi87mz04p068v1dd25wgs11v8pfl4-home-manager-generation.drv
building '/nix/store/56al72sri1bv11rg858m2f9s17j9sb6x-home-manager-path.drv'...
error: collision between `/nix/store/r7r10qvsqlnvbzjkjinvscjlahqbxifl-gcc-wrapper-11.3.0/bin/strip' and `/nix/store/2b99rpx8zwdjjqkrvk7kqgn9mxhiidjy-binutils-2.38/bin/strip'
error: builder for '/nix/store/56al72sri1bv11rg858m2f9s17j9sb6x-home-manager-path.drv' failed with exit code 25;
       last 1 log lines:
       > error: collision between `/nix/store/r7r10qvsqlnvbzjkjinvscjlahqbxifl-gcc-wrapper-11.3.0/bin/strip' and `/nix/store/2b99rpx8zwdjjqkrvk7kqgn9mxhiidjy-binutils-2.38/bin/strip'
       For full logs, run 'nix log /nix/store/56al72sri1bv11rg858m2f9s17j9sb6x-home-manager-path.drv'.
error: 1 dependencies of derivation '/nix/store/4fwfi87mz04p068v1dd25wgs11v8pfl4-home-manager-generation.drv' failed to build
```

What's odd is that the following works: `nix-shell '<home-manager>' -p home-manager clang` and `nix-shell '<home-manager>' -p home-manager gcc`.

**Comment by notfed** [comment]
I resolved this last week but don't know which step was key. I did all of the following:
1. Uninstalled home-manager with `nix-env --uninstall home-manager-path` 
2. Ran `nix-shell '<home-manager>' -p home-manager` to get a temporary home-manager
3. Uninstalled clang/gcc by removing them from `home-manager.nix`, and ran `home-manager switch`
4. Upgraded all my nix channels to nix-unstable
5. Haven't had the issue since
